### DISCUSSION TITLE
Add tile layer (XYZ) support in deck.gl maps

### DISCUSSION THREAD

**Discussion by fxprunayre** [apache/superset#27475] [Ideas]
In the DeckGL map plugin we can only select a Mapbox style from the [predefined list](https://github.com/apache/superset/blob/2499a1cf5a7f298c1ee2f34b3d67ca1d18bb7457/superset-frontend/plugins/legacy-preset-chart-deckgl/src/utilities/Shared_DeckGL.jsx#L373-L380) or [typing a new Mapbox style URL](https://github.com/apache/superset/pull/26031).

For users not having a Mapbox account, this would facilitate having a background layer on map charts.
 
Deck.gl [TileLayer](https://deck.gl/docs/api-reference/geo-layers/tile-layer) provides support for this and the well known [OpenStreeMap](https://www.openstreetmap.org/) layer provided by https://[abc].tile.openstreetmap.org/{z}/{x}/{y}.png could be used

![image](https://github.com/apache/superset/assets/1701393/d7a2cdd4-9f1e-4031-afc8-7537261298bb)

and would be a good alternative to the default white background layer

![image](https://github.com/apache/superset/assets/1701393/21728f1f-2d4e-4f07-840b-49aff7668fb0)

Other related work: https://github.com/apache/superset/pull/25869 also propose tile server support but in a specific cartodiagram plugin.

**Comment by ca5p3r** [comment]
Good find, sir.

We're currently using Superset internally for monitoring our spatial data consumption, and have modified the base DeckGL components to use OpenStreetMap or any tile service provider instead of MapBox for the same reason.

Would the Superset community be interested in such a feature? We're open to make a pull request with these updates, ditching MapBox entirely -only in DeckGL based charts- and replace it with a tile map service provided as environment variable or through superset_config.py

**Reply by dataappengineer** [reply]
Hello @ca5p3r,

I'm really excited about the possibility of using OpenStreetMap or other tile service providers instead of MapBox in DeckGL-based charts! This is exactly what our organization needs right now.

We're implementing Superset for a public sector project where we face strict requirements to use open source software exclusively. The current MapBox dependency has been a significant hurdle for us, as we cannot justify the commercial licensing costs within our open source mandate.

Your modified DeckGL components sound like the perfect solution to our problem. Would you be willing to share your implementation, either as a fork we could access or through the pull request you mentioned? Even a preview version would be incredibly helpful while waiting for the official PR to be processed.

I'd be happy to test your implementation in our environment and provide feedback on how it works with our specific use cases and datasets. This could potentially help strengthen the PR when it's submitted to the Superset community.

Thank you for considering this modification - it would make Superset much more accessible for organizations with open source requirements!

**Comment by BrendanBecaert** [comment]
Hello ! It looks like this PR : https://github.com/apache/superset/pull/33603 is implementing what you describe !

**Reply by ca5p3r** [reply]
Yes, it seems so! Despite having some merge issues, it would be great merging this feature.

Our current implementation is completely switching from MapBox since it doesn't fit to our Open-source policy.
